src/librustc_typeck/collect.rs

```diff
@@ -1377,20 +1407,21 @@ fn find_existential_constraints<'a, 'tcx>(
         }
     }
 
+    let node_id = tcx.hir().as_local_node_id(def_id).unwrap();
+    let outermost_id = tcx.hir().get_outermost_existential(node_id);
```

Comment:
Oh, now I understand this approach. The problem with the current impl is that the code now tries to get the concrete type of the outer existential type for the `impl Trait`. You need to still pass in the `def_id` of the `impl Trait`

Reply:
This is old code, sorry. I've fixed it now. I'm slow to push code, so maybe assume you don't need to review unless I request. Would hate to waste your time. :-)